mockk/src/main/kotlin/io/mockk/impl/annotations/JvmMockInitializer.kt

```diff
@@ -67,9 +78,113 @@ class JvmMockInitializer(val gateway: MockKGateway) : MockKGateway.MockInitializ
         }
     }
 
+    private fun resolveInstance(target: Any, property: KProperty1<Any, Any>): Any? {
+
+        var obj: Any?
+        val objCls = property.returnType.classifier as? KClass<*>
+
+        if (objCls != null) {
+            obj = objCls.objectInstance
+
+            if (obj == null) {
+                if (objCls.constructors.isEmpty() && objCls.primaryConstructor == null) {
+                    throw MockKException("No constructors found to instantiate @InjectMockK field ${property.name}")
+                } else {
+
+                    obj = resolveByConstructorInjection(target, objCls)
+
+                    if (obj == null) {
+                        try {
+                            val noArgCtor = objCls.constructors.first {
+                                it.parameters.isEmpty()
+                            }
+
+                            obj = noArgCtor.call()
+                            resolveByPropertyInjection(target, obj)
+                            (property as KMutableProperty1<Any, Any?>).set(target, obj)
+                        } catch (ex: NoSuchElementException) {
+                            throw MockKException("No suitable constructors found to instantiate @InjectMockK field ${property.name}")
+                        }
+                    } else {
+                        (property as KMutableProperty1<Any, Any?>).set(target, obj)
+                    }
+                }
+            } else {
+                throw MockKException("We could not inject mocks inside of @InjectMockK field ${property.name} because " +
+                        "the field already has a value.  Remove the field assignment, or provide your own dependencies.")
+            }
+        } else {
+            throw MockKException("Unable to create @InjectMockK field ${property.name} - could not cast it into a KClass type " +
+                    "required for instance creation.")
+        }
+
+        return obj
+    }
+
+    private fun resolveByConstructorInjection(target: Any, injectMockKCls: KClass<*>) : Any? {
+        // use largest constructor like mockito - maybe we can improve this part
+        val ctor = injectMockKCls.constructors.maxBy { it.parameters.size }
```

Comment:
you can `?: throw MockKException("no biggest constructor found")` to get rid of nullable type after